DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UnsupportedViewControllerRecorder.swift

```diff
@@ -0,0 +1,68 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+import SafariServices
+
+internal struct UnsupportedViewControllerRecorder: NodeRecorder {
+    // swiftlint:disable opening_brace
+    private let unsupportedViewControllersPredicates: [(UIViewController) -> Bool] = [
+        { $0 is SFSafariViewController },
+        { $0 is UIActivityViewController }
+    ]
+    // swiftlint:enable opening_brace
+
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) -> NodeSemantics? {
+        guard let viewController = view.parentViewController else {
+            return nil
+        }
+        guard unsupportedViewControllersPredicates.contains(where: { $0(viewController) }) else {
+            return nil
+        }
+        guard attributes.isVisible else {
+            return InvisibleElement(subtreeStrategy: .ignore)
+        }
+        let builder = UnsupportedViewControllerWireframesBuilder(
+            wireframeRect: view.frame,
+            wireframeID: context.ids.nodeID(for: view),
+            unsupportedClassName: String(reflecting: type(of: viewController)),
+            attributes: attributes
+        )
+        let node = Node(viewAttributes: attributes, wireframesBuilder: builder)
+        return SpecificElement(subtreeStrategy: .ignore, nodes: [node])
+    }
+}
+
+internal struct UnsupportedViewControllerWireframesBuilder: NodeWireframesBuilder {
+    let wireframeRect: CGRect
+
+    let wireframeID: WireframeID
+    let unsupportedClassName: String
+    let attributes: ViewAttributes
+
+    func buildWireframes(with builder: WireframesBuilder) -> [SRWireframe] {
+        return [
+            builder.createTextWireframe(
+                id: wireframeID,
+                frame: attributes.frame,
+                text: unsupportedClassName,
+                textFrame: attributes.frame,
+                textAlignment: .init(horizontal: .center, vertical: .center),
+                textColor: UIColor.red.cgColor,
+                borderColor: UIColor.lightGray.cgColor,
+                borderWidth: 1,
+                backgroundColor: UIColor(white: 0, alpha: 0.05).cgColor,
+                cornerRadius: 8
+            )
+        ]
+    }
+}
+
+fileprivate extension UIResponder {
+    var parentViewController: UIViewController? {
+        return next as? UIViewController ?? next?.parentViewController
+    }
+}
```

Comment:
I'm quite concerned about this. With that approach, recording complexity is no longer `O(N)` (N - the number of views in hierarchy), but worse 🐌. Here we traverse the responder chain backward during each step of forward view-tree traversal - it must be slower than `O(N)`.

I collected quick measures with landmarks example app, measuring the time spend in [`snapshotBuilder.createSnapshot(of: window, with: context)`](https://github.com/DataDog/dd-sdk-ios/blob/dd3fc8077342c90a84b825455cedb9a4a9a8e7f5/DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/WindowViewTreeSnapshotProducer.swift#L21) with `UnsupportedViewControllerRecorder` enabled and without it (using the [benchmark snippet](https://gist.github.com/ncreated/db931ed0202721b50fab56c3ccdcd4e8) made a weeks ago):
```
List screen:
* without: AVG createSnapshot(): 0.8357524871826172 ms
* with: AVG createSnapshot(): 1.0014202859666612 ms (+20%)

Modal screen:
* without: AVG createSnapshot(): 0.969586107465956 ms
* with: AVG createSnapshot(): 1.1173990037706163 ms  (+15%)

UI Controls screen:
* without: AVG createSnapshot(): 0.4487183358934191 ms
* with: AVG createSnapshot(): 0.5008896191914876 ms (+12%)

With calendar control opened:
* without: AVG createSnapshot(): 1.262982686360677 ms
* with: AVG createSnapshot(): 1.3359242015414767 ms (+5%)
```
It gets worse on more complex screens. As a baseline, I did same tests with disabling some other node recorders - none of them seems to bring +20% impact.

IMHO, to be performant, view controller check requires more presence in our architecture. What I can quickly propose is:
* computing responder chain snapshot separately (that would be `O(R)` complexity)
* adding sort of `context.responderChainSnapshot.isPartOfUnsupportedVC(view)` function that is `O(1)`

This way we will end up with `O(N + R)` complexity, so `O(N)` effectively.

Reply:
Got it. Very good point. I haven't considered the performance bottleneck here.

There other strategy I was thinking about is exctracting concrete private classes from those view controllers and ignoring them using previously added `UnsupportedViewsRecorder`

Reply:
If only we find it reliable, it could be a good short-term option 👍. Given that we will eventually support "unsupported view controllers" 😉, it might be a good choice.

On the other hand, I believe that "getting into VCs hierarchy" is unavoidable in the long term (e.g. might be required for fine-grained privacy controls). We could revisit the option I proposed once we find more needs for it.